samples/scala-reliable-timers/src/test/scala/com/example/actions/OrderActionSpec.scala

```diff
@@ -12,50 +10,16 @@ import org.scalatest.wordspec.AnyWordSpec
 // As long as this file exists it will not be overwritten: you can maintain it yourself,
 // or delete it so it is regenerated as needed.
 
-class OrderActionSpec
-    extends AnyWordSpec
-    with Matchers {
+class OrderActionSpec extends AnyWordSpec with Matchers with ScalaFutures {
 
   "OrderAction" must {
 
-    "have example test that can be removed" in {
-      val service = OrderActionTestKit(new OrderAction(_))
-      pending
-      // use the testkit to execute a command
-      // and verify final updated state:
-      // val result = service.someOperation(SomeRequest)
-      // verify the reply
-      // result.reply shouldBe expectedReply
-    }
-
     "handle command PlaceOrder" in {
       val service = OrderActionTestKit(new OrderAction(_))
-          pending
-      // val result = service.placeOrder(OrderRequest(...))
-    }
-
-    "handle command Confirm" in {
-      val service = OrderActionTestKit(new OrderAction(_))
-          pending
-      // val result = service.confirm(OrderNumber(...))
-    }
-
-    "handle command Cancel" in {
-      val service = OrderActionTestKit(new OrderAction(_))
-          pending
-      // val result = service.cancel(OrderNumber(...))
-    }
+      val result = service.placeOrder(OrderRequest(item = "Pizza Margherita", quantity = 3))
 
-    "handle command Expire" in {
-      val service = OrderActionTestKit(new OrderAction(_))
-          pending
-      // val result = service.expire(OrderNumber(...))
-    }
-
-    "handle command GetOrderStatus" in {
-      val service = OrderActionTestKit(new OrderAction(_))
-          pending
-      // val result = service.getOrderStatus(OrderNumber(...))
+      val timer = result.nextSingleTimerDetails[OrderNumber, Empty]
+      timer.name should startWith("order-expiration-timer")
```

Comment:
In this example, we can't test much more than this. The problem is that after setting a timer, we do a call to the Order entity component and component calls are not supported by the testkit. 

I'm not so convinced about timers support in the testkit anymore. The limitations we have for the testkit will very block the users in most of the cases.

Of course, we can have actions that simply add a timer, but I do expect the most often than not, timers will be used in combination with component calls. Making unit tests impossible.

The reason this test is passing is because the component call is done in the flatMap of the timer result. So, it doesn't interfere with the tests. 

The other usage of the timers API we have in this Order sample can't even be tested. We cancel the timer when confirming or cancelling an order. But we first call the entity and then cancel the timer. So, the test will fail when calling the component.

Reply:
in retrospect, I don't think that a method `nextSingleTimerDetails[In, Out]` makes much sense in the context here. 

`In/Out` are the types of the deferred call, not the timer. So maybe better to have an expected method in the `SingleTimerDetails` to check if the DeferredCall it holds has the expect type.